utils.go

```diff
@@ -25,22 +25,68 @@ func queryParseDefault[V QueryType](err error, parser func() V, defaultValue ...
 	return parser()
 }
 
-func queryParseInt[V QueryType](q string, bitSize int, parser func(int64) V, defaultValue ...V) V {
-	result, err := strconv.ParseInt(q, 10, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseInt[V GenericType](str string, bitSize int, parser func(int64) V, defaultValue ...V) V {
+	result, err := strconv.ParseInt(str, 10, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseUint[V QueryType](q string, bitSize int, parser func(uint64) V, defaultValue ...V) V {
-	result, err := strconv.ParseUint(q, 10, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseUint[V GenericType](str string, bitSize int, parser func(uint64) V, defaultValue ...V) V {
+	result, err := strconv.ParseUint(str, 10, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseFloat[V QueryType](q string, bitSize int, parser func(float64) V, defaultValue ...V) V {
-	result, err := strconv.ParseFloat(q, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseFloat[V GenericType](str string, bitSize int, parser func(float64) V, defaultValue ...V) V {
+	result, err := strconv.ParseFloat(str, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseBool[V QueryType](q string, parser func(bool) V, defaultValue ...V) V {
-	result, err := strconv.ParseBool(q)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseBool[V GenericType](str string, parser func(bool) V, defaultValue ...V) V {
+	result, err := strconv.ParseBool(str)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+}
+
+func genericParseType[V GenericType](str string, v V, defaultValue ...V) V {
+	switch any(v).(type) {
+	case int:
+		return genericParseInt[V](str, 32, func(i int64) V { return assertValueType[V, int](int(i)) }, defaultValue...)
```

Comment:
From strconv.ParseInt:
> // The bitSize argument specifies the integer type
// that the result must fit into. Bit sizes 0, 8, 16, 32, and 64
// correspond to int, int8, int16, int32, and int64.
// If bitSize is below 0 or above 64, an error is returned.

So that 32 should be changed to a 0

Reply:
thanks